core/templates/dev/head/pages/exploration_player/conversation_skin_directive.html

```diff
@@ -105,6 +105,10 @@
                                   style="margin-left: 10px;">
         </exploration-summary-tile>
       </div>
+
+      <div>
+        <a ng-href="/collection/{{collectionId}}" class="conversation-skin-back-to-collection" translate="I18N_PLAYER_BACK_TO_COLLECTION"></a>
```

Comment:
The approach seems reasonable on first glance. I haven't looked into detail yet though. You might need to double-check that $scope.collectionId does exist; an easy way to debug that is to just type `<[collectionId]>` somewhere on the page and see if anything shows up.